src/Type/Php/IsAFunctionTypeSpecifyingExtension.php

```diff
@@ -42,6 +46,16 @@ public function specifyTypes(FunctionReflection $functionReflection, FuncCall $n
 		$allowStringType = isset($node->getArgs()[2]) ? $scope->getType($node->getArgs()[2]->value) : new ConstantBooleanType(false);
 		$allowString = !$allowStringType->equals(new ConstantBooleanType(false));
 
+		// early exit in some cases, so we don't create false positives via IsAFunctionTypeSpecifyingHelper
+		if ($classType instanceof ClassStringType && !$classType instanceof GenericClassStringType) {
```

Comment:
especially this condition is pretty ugly .. would love feedback on how to improve it.